src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -487,8 +488,11 @@ export class DataResourceDataProvider implements IGridDataProvider {
 		// interface than the query execution service's saveResults handlers. Here, we take the
 		// format-specific request params (eg, includeHeaders for CSV) and merge the format-agnostic
 		// request params for the serialization request (eg, saveFormat, filePath).
+		let connProviders = this.cellModel.notebookModel.getApplicableConnectionProviderIds(this.cellModel.notebookModel.selectedKernelDisplayName);
```

Comment:
What if there's more than one provider? Wouldn't we have the same issue as we do currently in that case? Maybe we should at least log a warning.

Reply:
Doesn't look like we ever register more than one provider per kernel, so it might be something we can simplify

Reply:
Let me look around and see if there's a "current" connection provider in the notebookModel somewhere.